api/src/main/scala/dcos/metronome/api/v1/controllers/JobScheduleController.scala

```diff
@@ -76,12 +76,15 @@ class JobScheduleController(
     }
 
     withJobSpec(id) { spec =>
```

Comment:
the original IllegalArgumentException was throws few lines above here via a require method, just unwrap more of the code when reviewing :)

Reply:
I don't have a nice solution at hand but I think the flow could be simpler. We already pass along a function that performs the create/update/delete, we could as well wrap the predicate (must not exist yet/must exist etc) and pass that along.

Ideally we'd only do `jobSpecService.updateJobSpec` and handle either _JobSpec doesn't exist_ or _operation on schedule is not applicable_.